contents/security/sqlInjection.md

```diff
@@ -0,0 +1,145 @@
+# SQL Injection
+
+Author: [Lewis Koh](https://github.com/nus-cs3281/2018/blob/master/students/lewisKoh/lewisKoh-Resume.md)
+
+## Overview
+
+SQL (Structured Query Language) Injection is a vulnerability found in websites using databases.
+
+Text boxes are commonly used in forms in websites to ask the user for input. For example, 
+blog websites may provide a text box for owners of the blog to log in to the site. These 
+text boxes are used to query the underlying database of the website.
+
+However, are users limited to just typing in plain text? Some websites do not check the 
+input, thus the users are able to use escape strings to alter the query, or even to add 
+in their own. This can easily be used for malicious purposes. For example, users are able 
+to add parameters to the query.
+
+```html
+Username: foo
+Password: bar’ OR ‘’=’
+```
+
+The SQL command string built from this input would be as follows:
+
+```html
+SELECT Username FROM Users WHERE Username = ‘foo’
+AND Password = ‘bar’ OR ‘’=’’
+```
+
+This query will always return true, as ‘’=’’ is always true, and it is connected using an 
+OR comparator. As such, the query can be simplified to this:
+
+```html
+SELECT Username FROM Users
+```
+
+This will return all the rows from the Users table in the database, regardless of username 
+or password entered. If used in user authentication, the user will likely be able to gain
+access to the system.
+
+
+## Methods of SQL Injection
+
+There are two common methods for attackers to identify and exploit an SQL injection weakness
+in websites:
+
+* SQL Injection - The attacker makes use of errors generated from executed queries. The 
+errors from the database allows the attacker to fine-tune their attack so their next 
+attempt would be more successful.
+
+For example, if when attempting this attack to retrieve data from a separate table:
```

Comment:
emmm I was a bit confused when i saw the term "a separate table". Probably you can say, to retrieve data from table named "sysobjects"? do correct me if i m wrong orz